dashboard/src/app/dashboard/[dashboardId]/(dashboard)/WeeklyHeatmapSection.tsx

```diff
@@ -0,0 +1,165 @@
+  { value: 'session_duration', labelKey: 'sessionDuration' },
+] as const;
+
+export default function WeeklyHeatmapSection(props: WeeklyHeatmapSectionProps) {
+  const allData = use(props.weeklyHeatmapAllPromise);
+  const [selectedMetric, setSelectedMetric] = useState<HeatmapMetric>('unique_visitors');
+
+  const current: WeeklyHeatmapPrepared | undefined = useMemo(() => {
+    const pair = allData.find(([metric]) => metric === selectedMetric);
+    return pair ? pair[1] : undefined;
+  }, [allData, selectedMetric]);
+
+  const selectedMetricLabel = {
+    pageviews: 'Pageviews',
+    unique_visitors: 'Unique Visitors',
+    sessions: 'Sessions',
+    bounce_rate: 'Bounce Rate',
+    pages_per_session: 'Pages / Session',
+    session_duration: 'Session Duration',
+  }[selectedMetric];
+
+  const onMetricChange = (next: string) => {
+    setSelectedMetric(next as HeatmapMetric);
+  };
+
+  return (
+    <div className='bg-card border-border rounded-lg border p-6 shadow'>
+      <div className='mb-8 flex items-center justify-between'>
+        <div>
+          <h2 className='text-foreground mb-1 text-lg font-bold'>Weekly Trends</h2>
+        </div>
+        <div className='w-36'>
+          <Select value={selectedMetric} onValueChange={onMetricChange}>
+            <SelectTrigger className='w-full'>
+              <SelectValue />
+            </SelectTrigger>
+            <SelectContent>
+              {metricOptions.map((opt) => (
+                <SelectItem key={opt.value} value={opt.value}>
+                  {
+                    {
+                      pageviews: 'Pageviews',
+                      unique_visitors: 'Unique Visitors',
+                      sessions: 'Sessions',
+                      bounce_rate: 'Bounce Rate',
+                      pages_per_session: 'Pages / Session',
+                      session_duration: 'Session Duration',
+                    }[opt.value]
+                  }
+                </SelectItem>
+              ))}
+            </SelectContent>
+          </Select>
+        </div>
+      </div>
+
+      <HeatmapGrid
+        data={current?.matrix ?? []}
+        maxValue={current?.maxValue ?? 1}
+        metricLabel={selectedMetricLabel}
+        metric={selectedMetric}
+      />
+    </div>
+  );
+}
+
+type HeatmapGridProps = {
+  data: WeeklyHeatmapMatrix[];
+  maxValue: number;
+  metricLabel: string;
+  metric: HeatmapMetric;
+};
+
+function HeatmapGrid({ data, maxValue, metricLabel, metric }: HeatmapGridProps) {
+  const dayLabels = ['Mon.', 'Tue.', 'Wed.', 'Thu.', 'Fri.', 'Sat.', 'Sun.'];
+
+  const effectiveMax = Math.max(1, maxValue);
+
+  const getCellStyle = useCallback(
+    (value: number): CSSProperties => {
+      if (value <= 0) return {};
+
+      const t = Math.log1p(value) / Math.log1p(effectiveMax);
+      const eased = Math.pow(t, 0.85);
+
+      return {
+        backgroundColor: `oklch(62% 0.17 268.71 / ${0.1 + 0.9 * eased})`,
```

Comment:
Nitpick: Move the `0.1 + 0.9 * eased` to outside of the string interpolation for clarity/readability